### DISCUSSION TITLE
SDK build is nondeterministically failing to build

### DISCUSSION THREAD

**Discussion by james-a-morris** [ethereum-optimism/developers#309] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

cc @evaldofelipe 

We are currently experiencing a build issue in our CI. Occasionally - the `@eth-optimism/sdk` fails to build based on the pre-install script `only-allow pnpm`. We see that this has been updated in `https://github.com/ethereum-optimism/optimism/commit/3ccd12fe5c8c4c5a6acbf370d474ffa8db816562`. We also see that the `@eth-optimism/sdk` was bumped to `3.2.3`, which included the removal of the preinstall script.

However, we are now getting the same issue in a dependent package `@eth-optimism/bedrock-contracts@0.17.1`. It appears that `@eth-optimism/sdk@latest` uses `@eth-optimism/bedrock-contracts@0.17.1` which still fails.

Is it possible for the team to bump `@eth-optimism/bedrock-contracts` and then bump `@eth-optimism/sdk` so that we no longer have a failing build process?

### Additional Information

Our initial mitigation plan was to pre-install `only-allow` because we were initially getting an error related to `npx` being unable to find the `only-allow` binary.

```
# This command fix a concurrency issue when the package is not found.
RUN npx -y only-allow npm
```

**Comment by sbvegan** [comment]
Let me flag this to the maintainers of the SDK

**Comment by roninjin10** [comment]
This should be fixed if you upgrade to latest sdk https://github.com/ethereum-optimism/optimism/pull/9805

**Reply by james-a-morris** [reply]
Hey! Thanks for the quick reply. The SDK has a dependency on `@eth-optimism/bedrock-contracts` which has the `only-allow`.

**Comment by james-a-morris** [comment]
If possible - I'd like to keep this open given that [SDK@3.2.3](https://www.npmjs.com/package/@eth-optimism/sdk/v/3.2.3) still has a dependency on [op/bedrock-contracts](https://www.npmjs.com/package/@eth-optimism/contracts-bedrock/v/0.17.1) which has the pre-install script.
